ibm_catalog.json

```diff
@@ -0,0 +1,66 @@
+{
+  "products": [
+    {
+      "name": "terraform-ibm-secrets-manager-iam-engine",
+      "label": "IAM secrets engine module",
+      "product_kind": "module",
+      "tags": [
+        "dev_ops",
+        "target_terraform",
+        "terraform",
+        "module"
+      ],
+      "keywords": [
+        "terraform",
+        "iam",
+        "secrets",
+        "encryption"
+      ],
+      "short_description": "This module configures an IAM secrets engine for an existing secrets manager instance.",
+      "long_description": "Use this module to configure an IAM secrets engine for an existing secrets manager instance.\n#### Usage\n```hcl\nmodule \"iam_secrets_engine\" {\n  source                               = \"terraform-ibm-modules/terraform-ibm-secrets-manager-iam-engine/ibm\"\n  version                              = \"X.X.X\"  # Replace \"X.X.X\" with a release version to lock into a specific release\n  resource_group_id                    = \"xxXXxxXXxXxXXXXxxXxxxXXXXxXXXXX\"\n  region                               = \"us-south\"\n  iam_engine_name                      = \"iam-engine\"\n  secrets_manager_guid                 = \"XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX\"\n}\n```",
+      "offering_docs_url": "https://github.com/terraform-ibm-modules/terraform-ibm-secrets-manager-iam-engine/blob/main/README.md",
+      "offering_icon_url": "https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-secrets-manager-iam-engine/main/images/sm_icon.svg",
+      "features": [
+        {
+          "title": "Creates a new ServiceID and assigns role access",
+          "description": "Creates a new ServiceID. Assigns `Editor` role access for `iam-groups`. Assigns `Operator` role access for `iam-identity`. "
+        },
+        {
+          "title": "Creates a new apikey",
+          "description": "Creates a new apikey for the ServiceID."
+        },
+        {
+          "title": "Option to create a secret group",
+          "description": "Option to create a new secret group. Also, provisions to pass an existing secret group."
+        },
+        {
+          "title": "Creates a new arbitrary secret",
+          "description": "Creates a new arbitrary secret in the secret group with the generated ServiceID apikey value."
+        },
+        {
+          "title": "Creates an IAM engine",
+          "description": "Configures the secret manager instance with the IAM engine using the ServiceID apikey."
+        }
```

Comment:
```suggestion
      "features": [
        {
          "title": "Create a service ID with IAM roles",
          "description": "Creates a service ID. Assigns the Editor role to the `iam-groups` key. Assigns the Operator role to the `iam-identity` key. "
        },
        {
          "title": "Create an API key",
          "description": "Creates an API key for the Service ID."
        },
        {
          "title": "Create or reuse a secrets group",
          "description": "Create a secrets group or use an existing group."
        },
        {
          "title": "Create an arbitrary secret",
          "description": "Creates an arbitrary secret in the secret group with the generated API key value."
        },
        {
          "title": "Creates an secrets engine",
          "description": "Configures the Secrets Manager secrets engine to store the API key."
        }
```

These descriptions were confusing to me. (What is `iam-groups` and `iam_identity`?) Not sure I got the details right. Users should understand why we're featuring these things by the description that we include. We want to be specific about where the roles are being assigned, why we're promoting that it's an arbitrary secret here. I don't think there is an IAM engine, but I know there is a SM secrets engine.

There a lots of typos in the repo that we could fix in the main readme file and all variable description strings while we're working on this now. I found these: 
- `ServiceID` (service ID), 
- `secret manager` (Secrets Manager), 
- `apikey` or `API Key` (API key), 
- `iam` (IAM), 
- `secret engine` (secrets engine)

Reply:
I see similar role information in https://cloud.ibm.com/docs/secrets-manager?topic=secrets-manager-configure-iam-engine&interface=terraform#before-configure-iam-engine. If those are correct, you could use those phrases (for example, `Editor platform role on the xxx service`).